sql/expression/function/time.go

```diff
@@ -1247,3 +1249,86 @@ func (c CurrTimestamp) Eval(ctx *sql.Context, row sql.Row) (interface{}, error)
 func (c CurrTimestamp) WithChildren(children ...sql.Expression) (sql.Expression, error) {
 	return NoArgFuncWithChildren(c, children)
 }
+
+const maxCurrentTimestampPrecision = 6
+
+type CurrentTimestamp struct {
+	expression.UnaryExpression
+}
+
+var _ sql.FunctionExpression = (*CurrentTimestamp)(nil)
+
+func NewCurrentTimestamp(val sql.Expression) sql.Expression {
```

Comment:
This already exists as `NewCurrTimestamp`

Reply:
That one is used for no arguments.
I needed a way to throw error for `current_timestamp(null)` and have `current_timestamp()` still work.

Reply:
Should just have one function. Make it a FunctionN instead of a Function1, and handle the case where there is 1 v. 0 args